examples/core/src/main/java/com/adobe/cq/forms/core/components/it/models/impl/DemoComponentImpl.java

```diff
@@ -18,19 +18,84 @@
 
 import com.adobe.cq.export.json.ComponentExporter;
 import com.adobe.cq.export.json.ExporterConstants;
-import com.adobe.cq.forms.core.components.util.AbstractContainerImpl;
+import com.adobe.cq.export.json.SlingModelFilter;
+import com.adobe.cq.wcm.core.components.models.Container;
+import com.adobe.cq.wcm.core.components.models.ListItem;
+import com.adobe.cq.wcm.core.components.util.AbstractComponentImpl;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
 import org.apache.sling.api.SlingHttpServletRequest;
 import org.apache.sling.api.resource.Resource;
 import org.apache.sling.models.annotations.Exporter;
 import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.InjectionStrategy;
+import org.apache.sling.models.annotations.injectorspecific.OSGiService;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+import org.apache.sling.models.factory.ModelFactory;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Model(
-    adaptables = { SlingHttpServletRequest.class, Resource.class },
+    adaptables = { SlingHttpServletRequest.class },
     adapters = { ComponentExporter.class },
     resourceType = { "forms-components-examples/components/demo/component" })
 @Exporter(
     name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
     extensions = ExporterConstants.SLING_MODEL_EXTENSION)
-public class DemoComponentImpl extends AbstractContainerImpl {
+public class DemoComponentImpl extends AbstractComponentImpl implements Container {
 
+  @Self(injectionStrategy = InjectionStrategy.OPTIONAL)
+  protected SlingHttpServletRequest request;
+
+  protected List<Resource> filteredChildComponents;
+
+  @OSGiService
+  private SlingModelFilter slingModelFilter;
```

Comment:
Can you add a small runtime test case in cypress (ie) preview any of the existing form in preview layer ?